tests/validators/test_with_default.py

```diff
@@ -436,3 +436,82 @@ class Model:
 
     assert m2.int_list_with_default is not m1.int_list_with_default
     assert m2.str_dict_with_default is not m1.str_dict_with_default
+
+
+def test_default_value() -> None:
```

Comment:
are you sure this should all be one big test?

we should test for:
* a simple `with_default`
* both function types that support `default_value`
* another schema type that doesn't

Reply:
Okay I can split it up